### PR TITLE
Feature/obligation upload

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
Although there is 27 files changed the actual change is quite small.

It basically creates a request handler to retrieve the authorities obligations and display this on the view.

Unit and integration tests to finish up and to include
ObligationController
UploadObligationsViewModel
GetObligationScheme
UploadObligationsViewModelMap
GetSchemeObligation
SchemeObligationDataMap
GetSchemeObligationHandler
SchemeObligationUploadDataMap
GetSchemeObligationUpload

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMap.cs [hunk 1]
```diff
@@ -15,24 +15,39 @@ public UploadObligationsViewModel Map(UploadObligationsViewModelMapTransfer sour
...
             var model = new UploadObligationsViewModel(source.CompetentAuthority);
+            
+            SetErrors(source, model);
+
+            foreach (var schemeObligationData in source.ObligationData.OrderBy(o => o.SchemeName))
+            {
+                model.SchemeObligations.Add(new SchemeObligationViewModel()
+                {
+                    SchemeName = schemeObligationData.SchemeName,
+                    UpdateDate = schemeObligationData.UpdatedDate != null ? schemeObligationData.UpdatedDate.ToString() : "-"
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMap.cs:26]: The wireframe just displays it as normal without the GMT part, assume as the column description has GMT is it

File: src/EA.Weee.RequestHandlers/Mappings/SchemeObligationDataMap.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            Condition.Requires(source).IsNotNull();
+
+            var returnData = new List<SchemeObligationData>();
+
+            foreach (var scheme in source)
+            {
+                if (scheme.ObligationSchemes.Any())
+                {
+                    returnData.Add(new SchemeObligationData(scheme.SchemeName,
+                        scheme.ObligationSchemes.First().UpdatedDate,
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/Mappings/SchemeObligationDataMap.cs:24]: Only a single Updated Date for an ObligationScheme entity. It gets updated on that table if/when changes are made to the list of ObligationSchemeAmounts

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMap.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMap.cs:26]: I see, okay

File: src/EA.Weee.Web/Areas/Admin/Controllers/ObligationsController.cs [hunk 3]
```diff
@@ -66,16 +67,20 @@ public async Task<ActionResult> UploadObligations(CompetentAuthority authority,
...
-                SchemeObligationUploadData data = null;
+                List<SchemeObligationUploadErrorData> errorData = null;
                 if (id.HasValue)
                 {
-                    data = await client.SendAsync(User.GetAccessToken(), new GetSchemeObligationUpload(id.Value));
+                    errorData = await client.SendAsync(User.GetAccessToken(), new GetSchemeObligationUpload(id.Value));
                 }
 
+                var schemeObligationData =
+                    await client.SendAsync(User.GetAccessToken(), new GetSchemeObligation(authority, 2022));
```
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Web/Areas/Admin/Controllers/ObligationsController.cs:77]: okay

File: src/EA.Weee.RequestHandlers/Mappings/SchemeObligationDataMap.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.RequestHandlers/Mappings/SchemeObligationDataMap.cs:24]: okay

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMapTransfer.cs [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
         public CompetentAuthority CompetentAuthority { get; set; }
 
-        public SchemeObligationUploadData UploadData { get; set; }
+        public List<SchemeObligationUploadErrorData> ErrorData { get; set; }
+
+        public List<SchemeObligationData> ObligationData { get; set; }
+
+        public UploadObligationsViewModelMapTransfer()
+        {
+            ObligationData = new List<SchemeObligationData>();
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMapTransfer.cs:17]: Error Data can be null as in when the action is first executed there is no error id  so we use the fact it is null for creation of the model.

[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/UploadObligationsViewModelMapTransfer.cs:17]: okay,

**Pull request merged**